### PR TITLE
Feature/restructure package using interface

### PR DESCRIPTION
**Pull request by azmym**
Restructure the code and use struct & interfaces

### CODE REVIEW

File: domain/repository_test.go [hunk 1]
```diff
@@ -2,6 +2,7 @@ package domain
 
 import (
 	"github.com/stretchr/testify/assert"
+	"net/http"
```
Comment (maksadbek) [inline domain/repository_test.go:4]: nit: it's a good practice to sort out imports. See https://vyacheslavpryimak.medium.com/golang-imports-sorting-grouping-hell-and-painless-tool-to-solve-it-6c8448d1fdf7

**Review by maksadbek — APPROVED** [review]
LGTM

File: services/user_service.go [hunk 2]
```diff
@@ -5,6 +5,23 @@ import (
...
+type (
+	UserServiceInterface interface {
+		GetUserById(int64) (*domain.User, *utils.ApplicationError)
+	}
+	userService struct {
+	}
+)
+
+var (
+	UserService UserServiceInterface
```
Comment (maksadbek) [inline services/user_service.go]: I don't actually like creating an empty interface and then assigning a struct instance to it. You could just create the UserService:
```Go
var UserService = &userService{}
```
Later you can use it whenever as a `UserServiceInterface` as it implement this interface.

Comment (azmym) [reply services/user_service.go]: Done, please have a look

File: domain/repository_test.go (hunk 1 above)
Comment (azmym) [reply domain/repository_test.go:4]: Thanks for the advice

**Review by maksadbek — APPROVED** [review]
LGTM!

**Pull request merged**